extensions/pkg/util/shoot_clients.go

```diff
@@ -66,8 +66,9 @@ func NewShootClients(c client.Client, clientset kubernetes.Interface, gardenerCl
 	}
 }
 
-// NewClientForShoot returns the rest config and the client for the given shoot namespace. It first looks to use the endpoint given by
-// v1beta1constants.SecretNameGardenerInternal and, if it cannot find that, then from v1beta1constants.SecretNameGardener.
+// NewClientForShoot returns the rest config and the client for the given shoot namespace. It first looks to use the "internal" kubeconfig
+// (the one with in-cluster address) as in-cluster traffic is free of charge. If it cannot find that, then it fallbacks to the "external" kubeconfig
+// (the one with external DNS name or load balancer address) and this usually translates to egress traffic costs.
```

Comment:
```suggestion
// (the one with external DNS name or load balancer address) and this usually translates to egress traffic costs.
// It is possible to always use the external address by starting gardenlet with environment veriable GARDENER_SHOOT_CLIENT="external".
```

Reply:
I think the same info is already present in the line below (L73).

Reply:
True that. I missed it for some reason :facepalm: